test/ResponseHandlerTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace CalendArt\Adapter\Google;
+
+use GuzzleHttp\Message\ResponseInterface;
+
+use CalendArt\Adapter\Google\Exception;
+
+class ResponseHandlerTest extends \PHPUnit_Framework_TestCase
+{
+    private $response;
+    private $api;
+
+    protected function setUp()
+    {
+        $this->response = $this->prophesize(ResponseInterface::class);
+        $this->api = new Api;
+    }
+
+    public function testHandleErrorsWithSuccessfulResponse()
+    {
+        $this->response->getStatusCode()->shouldBeCalled()->willReturn(200);
+        $this->api->get($this->response->reveal());
+
+        $this->response->getStatusCode()->shouldBeCalled()->willReturn(301);
+        $this->api->get($this->response->reveal());
+    }
+
+    /**
+     * @dataProvider getResponses
+     */
+    public function testHandleErrors($statusCode, $reasonPhrase, $exception)
+    {
+        $this->setExpectedException($exception);
+
+        $this->response->getStatusCode()->shouldBeCalled()->willReturn($statusCode);
+        $this->response->json()->shouldBeCalled()->willReturn(['error' => ['message' => 'foo']]);
+        $this->response->getReasonPhrase()->willReturn($reasonPhrase);
+        $this->api->get($this->response->reveal());
+    }
+
+    public function getResponses()
+    {
+        return [
+            [400, 'Bad Request', Exception\BadRequestException::class],
+            [401, 'Invalid credentials', Exception\InvalidCredentialsException::class],
+            [403, 'Daily Limit Exceeded', Exception\DailyLimitExceededException::class],
+            [403, 'User Rate Limit Exceeded', Exception\UserRateLimitExceededException::class],
+            [403, 'Rate Limit Exceeded', Exception\RateLimitExceededException::class],
+            [403, 'Calendar usage limits exceeded.', Exception\CalendarUsageLimitsExceededException::class],
+            [404, 'Not Found', Exception\NotFoundException::class],
+            [409, 'The requested identifier already exists', Exception\IdentifierAlreadyExistsException::class],
+            [410, 'Gone', Exception\GoneException::class],
+            [412, 'Precondition Failed', Exception\PreconditionException::class],
+            [500, 'Backend Error', Exception\BackendException::class],
+        ];
+    }
+}
```

Comment:
instead of making a blank mock `Api` class, what you can do is using a mock that use the trait with `$this->getMockForTrait()` : https://phpunit.de/manual/current/en/test-doubles.html#test-doubles.mocking-traits-and-abstract-classes